Update compress.ts

thednp opened the pull request:
Add support for `font/woff2` compression. I will test this implementation.

### 🔗 Linked issue
None

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Sending a MonaSans(137k) / HubotSans(170k) file over to the client is quite expense no matter how you want to put it, we're expecting a 50-60% reduction in that size. Their OTF variants (currently supported by Nitro plugins) are out of the question with their more than double size.

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

manniL commented:
WOFF and WOFF2 are already compressed using Brotli. Compressing them again is not a good idea IMO

thednp commented:
So where should the woff files be located to get into compression pipeline?

manniL commented:
@thednp Nowhere, they are compressed by default already. No compression (again) needed as it won't save any bytes

thednp commented:
@manniL sorry to bother you guys, but I did some digging and found some interesting things you might wanna have a look at.
I modified the file at `node_modules/.pnpm/nitropack@2.8.1/node_modules/nitropack/dist/shared/nitro.4ea992bc.mjs` and added `  "font/woff",  "font/woff2",` and found the following:
* now my font IS included into the build compression pipeline
* the br file is exactly the same size as the original (137Kb), the gz file is 0.1Kb more for some strange reason
* see screenshot below

![Screenshot from 2024-01-15 20-44-57](https://github.com/unjs/nitro/assets/6761246/b4ce2a1f-da7a-4b14-8b37-0be61540db62)

May I suggest that you download a Woff2 file and make some tests yourselves, perhaps there's some bug you don't know yet about.

Thanks
